plus/src/main/java/org/apache/calcite/adapter/os/CpuInfoTableFunction.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.os;
+
+import org.apache.calcite.DataContext;
+import org.apache.calcite.linq4j.AbstractEnumerable;
+import org.apache.calcite.linq4j.Enumerable;
+import org.apache.calcite.linq4j.Enumerator;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.ScannableTable;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+/**
+ * Table function that executes the OS "cpu_info".
+ */
+public class CpuInfoTableFunction extends AbstractBaseScannableTable {
+  private CpuInfoTableFunction() {
+  }
```

Comment:
You should not make `CpuInfoTableFunction ` extend `AbstractBaseScannableTable`, return the `AbstractBaseScannableTable` instance in the eval method seems more reasonable.

Reply:
> You should not make `CpuInfoTableFunction ` extend `AbstractBaseScannableTable`, return the `AbstractBaseScannableTable` instance in the eval method seems more reasonable.

hi @danny0405  well, I've already changed it.